rotkehlchen/data_import/importers/bitcoin_tax.py

```diff
@@ -0,0 +1,337 @@
+        return hashlib.sha256(row_str).hexdigest()
+
+    @staticmethod
+    def determine_csv_type(csv_data: csv.DictReader) -> str:
+        if csv_data.fieldnames is None:
+            raise InputError('Empty BitcoinTax CSV file')
+        if 'Cost/Proceeds' in csv_data.fieldnames:
+            return 'trades'
+        return 'other_events'
+
+    def _consume_trade_event(
+            self,
+            cursor: DBCursor,
+            csv_row: dict[str, Any],
+            event_identifier: str,
+            timestamp: TimestampMS,
+            location: Location,
+            action: str,
+            base_asset: AssetWithOracles,
+            base_asset_amount: AssetAmount,
+            quote_asset: AssetWithOracles,
+            quote_asset_amount: AssetAmount,
+            fee_asset: Optional[AssetWithOracles],
+            fee_amount: Fee,
+            memo: str,
+    ) -> None:
+        if action not in ('BUY', 'SELL', 'SWAP'):
+            raise UnsupportedCSVEntry(f'Unsupported entry {action}. Data: {csv_row}')
+
+        events: list[HistoryBaseEntry] = []
+        event_type = HistoryEventType.TRADE
+
+        if action == 'BUY':
+            receive_asset = base_asset
+            receive_amount = base_asset_amount
+            spend_asset = quote_asset
+            spend_amount = quote_asset_amount
+        elif action == 'SELL':
+            receive_asset = quote_asset
+            receive_amount = quote_asset_amount
+            spend_asset = base_asset
+            spend_amount = base_asset_amount
+        else:  # action = 'SWAP'
+            # The SWAP action is used when a crypto asset is renamed or forked.
+            # It is assumed that rotki knows if the new asset has been renamed or forked.
+            # We detect if it is a rename or a fork and handle the fork case with a
+            # new receive history event. In case of rename nothing needs to be done.
+            event_type = HistoryEventType.RECEIVE
+            try:
+                receive_asset = quote_asset.resolve_to_crypto_asset()
+            except UnknownAsset:
+                log.info(
+                    f'Ignoring unknown crypto asset {quote_asset} in BitcoinTax CSV SWAP action')  # noqa: E501
+                return  # skip this csv row
+
+            if receive_asset.forked is None:
+                log.info(
+                    f'Ignoring renamed crypto asset {receive_asset} in BitcoinTax CSV SWAP action')  # noqa: E501
+                return  # skip this csv row
+
+            receive_amount = quote_asset_amount
+            receive_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=0,
+                timestamp=timestamp,
+                location=location,
+                asset=receive_asset,
+                balance=Balance(
+                    amount=receive_amount,
+                    usd_value=ZERO,
+                ),
+                notes=memo,
```

Comment:
memo is just an identifier, right? I am not sure if users want to see it in the notes. But no matter whether to keep it in the notes or not, notes should be more descriptive. Like "Spend `amount` `asset`, as imported from bitcoin tax"

Reply:
memo is an optional description for an entry (trade, income, expense) in bitcoin tax that has been created by the user, so I think that if we are to store it we should not change it but store it as it is in the csv so that the user can identify it if he needs to. So, I guess either store it in notes as it is or not at all?

Reply:
Ah, if memo is some sort of descriptive text, then it's fine. I would do then something like `f'Imported from bitcoin tax: {memo}'`